lib/loader.js

```diff
@@ -16,10 +16,14 @@ module.exports = function(content, map) {
 	var root = query.root;
 	var moduleMode = query.modules || query.module;
 	var camelCaseKeys = query.camelCase || query.camelcase;
+	var sourceMap = query.sourceMap || false;
 	var resolve = createResolver(query.alias);
 
-	if(map !== null && typeof map !== "string") {
+	if(sourceMap && map !== null && typeof map !== "string") {
 		map = JSON.stringify(map);
+	} else {
+        // Some loaders (example `"postcss-loader": "1.x.x"`) always generates source map, we should remove it
+		map = null;
```

Comment:
@michael-ciniawsky As I understood correctly, we should generally drop `sourceMap` option? and see on `map` from other loaders?